apps/dashboard/src/components/pay/PayAnalytics/hooks/usePayNewCustomers.ts

```diff
@@ -0,0 +1,64 @@
+import { useQuery } from "@tanstack/react-query";
+import { useLoggedInUser } from "../../../../@3rdweb-sdk/react/hooks/useLoggedInUser";
+import { THIRDWEB_PAY_DOMAIN } from "../../../../constants/urls";
+
+export type PayNewCustomersData = {
+  intervalType: "day" | "week";
+  intervalResults: Array<{
+    /**
+     * Date formatted in ISO 8601 format
+     */
+    interval: string;
+    distinctCustomers: number;
+  }>;
+  aggregate: {
+    // totals in the [fromDate, toDate] range
+    distinctCustomers: number;
+    bpsIncreaseFromPriorRange: number;
+  };
+};
+
+type Response = {
+  result: {
+    data: PayNewCustomersData;
+  };
+};
+
+export function usePayNewCustomers(options: {
+  clientId: string;
+  from: Date;
+  to: Date;
+  intervalType: "day" | "week";
+}) {
+  const { user, isLoggedIn } = useLoggedInUser();
+
+  return useQuery(
+    ["usePayNewCustomers", user?.address, options],
+    async () => {
+      const endpoint = new URL(
+        `https://${THIRDWEB_PAY_DOMAIN}/stats/aggregate/customers/v1`,
+      );
+      endpoint.searchParams.append("intervalType", options.intervalType);
+      endpoint.searchParams.append("clientId", options.clientId);
+      endpoint.searchParams.append("fromDate", `${options.from.getTime()}`);
+      endpoint.searchParams.append("toDate", `${options.to.getTime()}`);
+
+      const res = await fetch(endpoint.toString(), {
+        method: "GET",
+        credentials: "include",
+        headers: {
+          "Content-Type": "application/json",
+        },
+      });
+
+      if (!res.ok) {
+        throw new Error("Failed to fetch new customers");
+      }
+
+      const resJSON = (await res.json()) as Response;
+
+      return resJSON.result.data;
+    },
+    { enabled: !!user?.address && isLoggedIn },
```

Comment:
I *believe* `user.address` is only set in `useLoggedInUser` if the user is logged in and not before that, in which case this is redundant (but will work obv)

Reply:
removing redudancy